src/Components/Components/src/Sections/SectionContent.cs

```diff
@@ -37,11 +44,54 @@ void IComponent.Attach(RenderHandle renderHandle)
     }
 
     Task IComponent.SetParametersAsync(ParameterView parameters)
+    {
+        // We are not using parameters.SetParameterProperties(this)
+        // because IsDefaultContent is internal property and not a parameter
+        SetCustomParameters(parameters);
+
+        if (SectionName is not null && SectionId is not null)
+        {
+            throw new InvalidOperationException($"Both '{nameof(SectionName)}' and '{nameof(SectionId)}' cannot have values.");
+        }
+        else if (SectionName is not null)
+        {
+            _identifier = SectionName;
+        }
+        else if (SectionId is not null)
+        {
+            _identifier = SectionId;
+        }
+        else
+        {
+            throw new InvalidOperationException($"{nameof(SectionContent)} requires a non-null value either for '{nameof(SectionName)}' or '{nameof(SectionId)}'.");
+        }
+
+        if (!object.Equals(_identifier, _registeredIdentifier) || IsDefaultContent != _registeredIsDefaultContent)
+        {
+            if (_registeredIdentifier is not null)
+            {
+                _registry.RemoveProvider(_registeredIdentifier, this);
+            }
+
+            _registry.AddProvider(_identifier, this, IsDefaultContent);
+            _registeredIdentifier = SectionId;
```

Comment:
Now that you also can use `SectionName` to identify a section `_registeredIdentifier` should be set to `identifier` instead of `SectionId`. Otherwise when using the `SectionName` it would be null, and RemoveProvider isn't called when it should.

The same issue exists in SectionOutlet.

Reply:
@Flachdachs this is a closed PR, so could you log an issue with details around your concern?

Reply:
Of course. Actually my intention was to open an issue. I visited the page with the changes anonymously. It offered a "New Issue" button. But as I opened the same URL as logged in user the button was replaced with a Create Review button.